### PR TITLE
CB-25-Detailed-recipe-view

### PR DESCRIPTION
**Pull request by banaskrzysztof**
1. After clicking on recipe user should see detailed view
2. Should be possible to go back to list view
3. Url should change (recipe/:id)

### CODE REVIEW

File: src/app/modules/recipe-detail/containers/recipe-detail.component.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      margin: 5px;
+      content: url("https://1.bp.blogspot.com/-uLQuV_Nv7pg/XgDjRFSWsoI/AAAAAAAAT9c/tkM2CPXDomgInv25hXF9VsbozIgmhdAuQCNcBGAsYHQ/s640/79703750_451330148869455_7787295256956895232_n.jpg");
+      border-radius: 5px;
+    }
+
+    &__info {
+      padding: 5px;
+      font-size: large;
+
+      &__name {
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.scss:48]: delete empy slector

File: src/app/modules/recipe-detail/containers/recipe-detail.component.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+      }
+
+      &__dtr {
+        display: flex;
+        justify-content: space-between;
+      }
+
+      &__ingredients {
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.scss:57]: delete empty selector

File: src/app/modules/recipe-detail/containers/recipe-detail.component.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    this.activeRoute.params.subscribe(params => {
+      this.getRecipe(params['id']);
+    });
+
+  }
+
+  public getRecipe(id: number): void {
+    this.recipeDetailService.getRecipe(id).subscribe(recipe => {
+      this.recipe = recipe;
+      console.log(recipe);
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.ts:30]: remove console.log

File: src/app/modules/recipe-detail/recipe-detail.module.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import {CommonModule} from "@angular/common";
+import {RouterModule} from "@angular/router";
+import {RecipeDetailComponent} from "./containers/recipe-detail.component";
+import {MatIconModule} from "@angular/material/icon";
+
+
+@NgModule({
+  imports: [CommonModule,
+    RouterModule.forChild([
+      {path: '', component: RecipeDetailComponent}
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/recipe-detail.module.ts:11]: please create `const` with routing and move values to it
`export const RecipeDetailsRouting = [{path: '', component: RecipeDetailComponent}];`

File: src/app/modules/recipe-detail/services/recipe-detail.service.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {Observable} from "rxjs";
+
+import {Recipe} from "../../recipe-list/models/recipie.model";
+
+@Injectable({
+  providedIn: 'root'
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/services/recipe-detail.service.ts:8]: remove `providedIn`, and add `providers: [RecipeDetailsService]` to RecipeDetailModule

File: src/app/modules/recipe-list/containers/recipe-list.component.scss [hunk 6]
```diff
@@ -43,5 +46,9 @@
       border-bottom-style: solid;
       border-color: black;
     }
+
+    &__comment:hover {
+      background: #f2f2f2;
```
Comment (Bopiotr) [inline src/app/modules/recipe-list/containers/recipe-list.component.scss:51]: pls create css variable with this color

**Review by Bopiotr — CHANGES_REQUESTED** [review]

File: src/app/modules/recipe-list/containers/recipe-list.component.scss [hunk 7]
```diff
@@ -1,3 +1,5 @@
+$background: #f2f2f2;
```
Comment (Bopiotr) [inline src/app/modules/recipe-list/containers/recipe-list.component.scss:1]:
```suggestion
$background-color: #f2f2f2;
```

File: src/app/modules/recipe-detail/containers/recipe-detail.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+@Component({
+  selector: 'app-recipe-detail',
+  templateUrl: './recipe-detail.component.html',
+  styleUrls: ['./recipe-detail.component.scss']
+})
+export class RecipeDetailComponent implements OnInit {
+
+  public recipe: Recipe = {id: 0, name: '', content: ''};
+  id?: number;
```
Comment (Bopiotr) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.ts:14]:
```suggestion
  public id?: number;
```

**Review by Bopiotr — APPROVED** [review]

**Review by KrzysiekLipiec — APPROVED** [review]

**Pull request merged**